### PR TITLE
#475 Updates to OrgJoin template

### PR DESCRIPTION
**Pull request by CarlosNZ**
This pretty much completes [Epic #42 - Join organisation](https://github.com/openmsupply/application-manager-web-app/issues/475)

Requires front-end branch from [PR#498](https://github.com/openmsupply/application-manager-web-app/pull/498)

Now the Company selection in the front-end is carried through and applied correctly to join the user to organisation in the back-end on Review approval.

### To test:

(Requires fresh `database_init`)
1. Start a new OrgJoin application: http://localhost:3000/application/new?type=OrgJoin
2. Fill out and Submit application
3. Assign the user "Screener1" to this application using the following SQL code:  
```
UPDATE review_assignment SET status = 'Assigned' WHERE id = 1;
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '5002', '1');
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '43', '1');
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '42', '1');
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '39', '1');
```
4. As "Screener1", start review and approve all questions. Submit review.
5. Back-end should correctly update Status, Outcome, create a new user_organisation record, and grant "ApplyDrugReg" permissions to the user/org

Unrelated changes: added some missing documentation for "CleanupFiles" action

### CODE REVIEW

**Comment by nmadruga** [comment]
Oh! I haven't seen you already had the Back-end for [PR#498](https://github.com/openmsupply/application-manager-web-app/pull/498) here. I'll have a look on this tomorrow morning - first thing!

File: database/insertData/03_template_F_companyJoin.js [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+                  ]
+                }
+                parameterQueries: {
+                  applicationId: {
+                    operator: "objectProperties"
+                    children: ["applicationData.applicationId"]
+                  }
+                  newStatus: "Completed"
+                }
+              }
```
Comment (nmadruga) [inline database/insertData/03_template_F_companyJoin.js:285]: Out of order - can move to sequential place please?

File: database/insertData/03_template_F_companyJoin.js [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+            ]
+          }
+          templatePermissionsUsingId: {
+            create: [
+              # applyJoinCompany
+              { permissionNameId: 8000 }
```
Comment (nmadruga) [inline database/insertData/03_template_F_companyJoin.js:291]:
```suggestion
              # applyJoinCompany
              { permissionNameId: 8500 }
```

Or - can remove the new PermissionName: 8500 and use the applyGeneral:
```
// applyGeneral (associated with multiple templates)
  `mutation assignGeneralPermission {
    createPermissionName(
      input: {
        permissionName: { 
          id: 10100
          name: "applyGeneral",  permissionPolicyId: 2000 }
      }
    ) {
      permissionName {
        name
      }
    }
  }`,
```

Comment (nmadruga) [reply database/insertData/03_template_F_companyJoin.js:291]: Also remember to add one action on the UserRegistrationTemplate to grant this permission to new user registered.

File: documentation/List-of-Action-plugins.md [hunk 3]
```diff
@@ -217,6 +217,16 @@ When an applicant re-submits an application after making changes, this Action up
...
+### Clean Up Files
+
+After application submission or re-submission, this Action looks at all the uploaded files associated with the application and deletes any that are not part of a submitted response. This takes care of any files that the user "deleted" from the UI before submitting.
+
+- _Action Code:_ **`cleanupFiles`**
+
+| Input parameters<br />(\*required) <br/> | Output properties |
+| ---------------------------------------- | ----------------- |
+| `applicationSerial`                      | `deletedFiles`    |
+
```
Comment (nmadruga) [inline documentation/List-of-Action-plugins.md:229]: Nice - is this one already running for all applications that are using core actions?

File: database/insertData/02_permission_names.js [hunk 4]
```diff
@@ -138,6 +138,20 @@ exports.queries = [
...
+        permissionName: { 
+          id: 8500
+          name: "applyJoinCompany", permissionPolicyId: 2000 }
+      }
+    ) {
+      permissionName {
+        name
+      }
+    }
+  }`,
```
Comment (nmadruga) [inline database/insertData/02_permission_names.js:154]: Maybe can be removed - if we only use the general apply for linking new user to this core template

**Review by nmadruga — CHANGES_REQUESTED** [review]
First run on this - I'll now test using the Front-end branch. 

Basically my suggestion is to use the GeneralApply and add an action to grant this to every new user registered. So we will have this feature of registering and being able to connect with a company after submitting this application : )

**Comment by CarlosNZ** [comment]
> Basically my suggestion is to use the GeneralApply and add an action to grant this to every new user registered

Yeah, this would be a change to UserRegistration template, so should probably do that in separate issue I think.

File: database/insertData/03_template_F_companyJoin.js (hunk 1 above)
Comment (CarlosNZ) [reply database/insertData/03_template_F_companyJoin.js:285]: Sure, done c28a1c66

File: documentation/List-of-Action-plugins.md (hunk 3 above)
Comment (CarlosNZ) [reply documentation/List-of-Action-plugins.md:229]: Yes, I think so.

**Comment by CarlosNZ** [comment]
Hey @nmadruga -- just FYI, this PR was purely meant to demonstrate the changes made in [front-end PR #458](https://github.com/openmsupply/application-manager-web-app/pull/498) and nothing else. I'm in the process of making more "correct" changes to the createOrg and joinOrg templates in later PRs (after additional functionality is added is other PRs I've already published), so can we take care of most of your permission-related suggestions then?

**Comment by nmadruga** [comment]
> I'm in the process of making more "correct" changes to the createOrg and joinOrg templates in later PRs (after additional functionality is added is other PRs I've already published), so can we take care of most of your permission-related suggestions then?

No problem : )

**Comment by CarlosNZ** [comment]
Closing -- superceded by #303

**Pull request closed without merging**